src/main/java/com/mathworks/ci/actions/RunMatlabBuildAction.java

```diff
@@ -0,0 +1,100 @@
+package com.mathworks.ci.actions;
+
+/**
+ * Copyright 2024, The MathWorks Inc.
+ *
+ */
+
+import java.io.File;
+import java.io.IOException;
+
+import hudson.FilePath;
+import hudson.model.Run;
+import hudson.console.LineTransformationOutputStream;
+
+import com.mathworks.ci.BuildArtifactAction;
+import com.mathworks.ci.BuildConsoleAnnotator;
+import com.mathworks.ci.MatlabExecutionException;
+import com.mathworks.ci.parameters.BuildActionParameters;
+import com.mathworks.ci.utilities.MatlabCommandRunner;
+
+public class RunMatlabBuildAction {
+    private BuildActionParameters params; 
+    private MatlabCommandRunner runner;
+    private BuildConsoleAnnotator annotator;
+
+    private static String DEFAULT_PLUGIN = 
+        "+ciplugins/+jenkins/getDefaultPlugins.m";
+    private static String BUILD_REPORT_PLUGIN = 
+        "+ciplugins/+jenkins/BuildReportPlugin.m";
+    private static String TASK_RUN_PROGRESS_PLUGIN = 
+        "+ciplugins/+jenkins/TaskRunProgressPlugin.m";
+
+    public RunMatlabBuildAction(MatlabCommandRunner runner, BuildConsoleAnnotator annotator, BuildActionParameters params) {
+        this.runner = runner;
+        this.annotator = annotator;
+        this.params = params;
+    }
+
+    public RunMatlabBuildAction(BuildActionParameters params) throws IOException, InterruptedException {
+        this(new MatlabCommandRunner(params), 
+                new BuildConsoleAnnotator(
+                    params.getTaskListener().getLogger(), 
+                    params.getBuild().getCharset()),
+                params);
+    }
+
+    public void run() throws IOException, InterruptedException, MatlabExecutionException {
+        // Copy plugins and override default plugins function
+        runner.copyFileToTempFolder(DEFAULT_PLUGIN, DEFAULT_PLUGIN);
+        runner.copyFileToTempFolder(BUILD_REPORT_PLUGIN, BUILD_REPORT_PLUGIN);
+        runner.copyFileToTempFolder(TASK_RUN_PROGRESS_PLUGIN, TASK_RUN_PROGRESS_PLUGIN);
```

Comment:
Great suggestion! I'll switch that.

Reply:
Hmm, this might actually be a little bit trickier than I thought before. It doesn't seem like there's any great ways to copy a folder out of the jar resources, so I think the change would probably require us to zip the entire folder up and then unzip it in that location.

I'm not sure that's worth it, as I think the change to zipping and unzipping would take up more space than just having the files listed out (unless we start adding a bunch more plugins.) What's your opinion on leaving it the way it is for now?

Reply:
Yeah leaving it this way is okay.